soc/arm/nordic_nrf/nrf91/CMakeLists.txt

```diff
@@ -5,3 +5,14 @@ zephyr_sources(
   )
 
 zephyr_sources_ifdef(CONFIG_ARM_MPU mpu_regions.c)
+
+if(DEFINED CONFIG_ARM_SECURE_FIRMWARE AND DEFINED CONFIG_ARM_FIRMWARE_HAS_SECURE_ENTRY_FUNCS)
+  if (NOT ${CONFIG_ARM_NSC_REGION_BASE_ADDRESS} EQUAL 0)
+    math(EXPR nsc_size "0x8000 - (${CONFIG_ARM_NSC_REGION_BASE_ADDRESS} % 0x8000)")
+    set(valid_sizes 32 64 128 256 512 1024 2048 4096)
+    if(NOT nsc_size IN_LIST valid_sizes)
+      message(WARNING "CONFIG_ARM_NSC_REGION_BASE_ADDRESS set to an address (${CONFIG_ARM_NSC_REGION_BASE_ADDRESS}) "
```

Comment:
@oyvindronningstad, @SebastianBoe  why don't we change this to FATAL_ERROR?

Reply:
I assume he wants to avoid a CMake error, as that would prevent generation, and thereby prevent menuconfig to be run to rectify the error.

Can get an error, and also menuconfig by moving this to C/LD. Or by limiting the choices in Kconfig.

Reply:
@SebastianBoe thanks.
So I would like to get an error here, if possible. I don't want the CMake to build with an erroneous NSC starting address.

Reply:
But, I guess, I _do_ want menuconfig, to fix that :)

Reply:
You can't get an error there and also have menuconfig, but we can do one of the options proposed.

Reply:
>We could get an error, and also menuconfig, by moving this to C/LD. Or by limiting the choices in Kconfig.

Don't get this fully, I am afraid. Pls, elaborate a bit more

Reply:
We can get an error from the C preprocessor through the use of #error, if we move this business logic into either a C or LD-script file.

We can also limit the users choices so that it is impossible to create a user-error. By introducing a choice symbol with a hardcoded set of choices, just as we have a hardcoded set of valid_sizes today.

Which one makes sense depends on how this option is used, probably it makes most sense to use the preprocessor.

Reply:
Yes, now I understand what you meant - sure, let's move this in pre-processor then.
Should we still keep this, here, as a warning, though?

Reply:
I'd remove it from here as it looks like that will need updating soon.

Reply:
ok, we can then remove it
I 've been asking for it, actually, sorry for the noise.

Reply:
Which source file should we add the check to do you think?

Reply:
Will remove. Which source file should we add the check to do you think?

Reply:
I think this should be in the file we actually configure the IDAU (so not present upstream, at the moment)

Reply:
Removed

Reply:
@SebastianBoe @ioannisg I just discovered that it is possible to ASSERT in linker scripts.

https://sourceware.org/binutils/docs-2.20/ld/Miscellaneous-Commands.html

I'll add a check using that.

Reply:
Added